### ISSUE TITLE
Loss rises during training。

### ISSUE THREAD

**Issue by zhiyuanjiang** [Tiger101010/DAEGC#1]
when i run your model, loss and acc are keep increasing, just like the picture below. 
![image](https://user-images.githubusercontent.com/28390107/120890819-acd3e180-c637-11eb-90be-a1b141c3adf3.png)

the model users the gradient descent method to optimize, but the loss is increased. I know that in the KL
 loss, the Q distribution is approached to the P distribution, and both distributions are changed during training. So i want to know whether it is normal for loss and acc to increase during training, and why?

**Comment by Tiger101010** [comment]
In my opinion, a potential reason is that the reconstruction loss has already converged in the pretrain stage, and then the joint learning stage refines the embedding learning to benefit clustring, however, which may not be beneficial to the reconstrustion job. 

As the authors mentioned in their paper, you can do some visualization to illustrate this process. Here are some embedding visualization below. The first one is from pretrain stage, and the second one is from joint learning stage.
![30](https://user-images.githubusercontent.com/34651180/120920038-60e47380-c6ef-11eb-8a33-a700112ee64d.png)
![46-1](https://user-images.githubusercontent.com/34651180/120920039-6215a080-c6ef-11eb-839e-bb462783a5ec.png)

**Comment by zhiyuanjiang** [comment]
Thanks for your answer, but clusterng loss is also increased.

**Comment by zhiyuanjiang** [comment]
Thanks for your answer, but clustering loss is also increased.

**Comment by zhiyuanjiang** [comment]
Hi, I still have a problem.  In the code below:
```python
        attn_for_self = torch.mm(h, self.a_self)  # (N,1)
        attn_for_neighs = torch.mm(h, self.a_neighs)  # (N,1)
        attn_dense = attn_for_self + torch.transpose(attn_for_neighs, 0, 1)  # 妙啊
        attn_dense = torch.mul(attn_dense, M)
        attn_dense = self.leakyrelu(attn_dense)  # (N,N)

        zero_vec = -9e15 * torch.ones_like(adj)
        adj = torch.where(adj > 0, attn_dense, zero_vec)
        attention = F.softmax(adj, dim=1)
```
according to the paper, we need to use generalized neighbors of node, so in code **adj = torch.where(adj > 0, attn_dense, zero_vec)**, why use **adj**, instead of **attn_dense**. just like:
```python
adj = torch.where(attn_dense > 0, attn_dense, zero_vec)
```

**Comment by Tiger101010** [comment]
`attn_dense` contains all conbination, but we only need those that acctually have edges. You can check comments in this [file](https://github.com/Diego999/pyGAT/blob/master/layers.py) of the [repo](https://github.com/Diego999/pyGAT) for detail.

**Comment by zhiyuanjiang** [comment]
Thanks for your answer. The paper proposes to exploit high-order neighbors. but `adj` is one-hop neighboring nodes, and `attn_dense` is high-hop neighboring nodes, isn't `attn_dense` more accurate?
